pkg/controller/controller.go

```diff
@@ -739,6 +780,65 @@ func findServicePlan(name string, plans []v1alpha1.ServicePlan) *v1alpha1.Servic
 	return nil
 }
 
+// handleLastOperationResponse gets called when a last operation poll has happened.
+func (c *controller) handleLastOperationResponse(instance *v1alpha1.Instance, resp *brokerapi.LastOperationResponse) bool {
+	switch resp.State {
+	case "in progress":
+		c.setAsyncOperationOngoing(instance, true)
+		return true
+	case "succeeded":
+		c.setAsyncOperationOngoing(instance, false)
+		// TODO(vaikas): Update the instance condition ready to true and set to succeed
+	case "failed":
+		c.setAsyncOperationOngoing(instance, false)
+		// TODO(vaikas): Update the instance condition ready to true and set to fail
+	default:
+		glog.Warningf("Got invalid state in LastOperationResponse: %q", resp.State)
+	}
+	return false
+}
+
+// setAsyncOperationOngoing updates the
+// InstanceConditionAsyncOperationInProgress to the specified status.
+func (c *controller) setAsyncOperationOngoing(instance *v1alpha1.Instance, status bool) {
+	if status {
+		c.updateInstanceCondition(
```

Comment:
@jwforres and @pmorie and I just a chat. There was a misunderstanding that Operation is always there, which is not the case according to OSB spec as well as the CF documents. Hence we can NOT use the operation as state for knowing that an async operation is ongoing against a ServiceInstance.
So, we need to store state about this. We can store it in the condition like I had done, but it was decided that instead of storing that state in a Condition, we'll add another field in the InstanceStatus indicating that an async operation is in progress. I'm fine with this approach and I'll change the code accordingly.
@arschles do you have any concerns with this approach?